specifyweb/frontend/js_src/lib/components/UserPreferences/index.tsx

```diff
@@ -129,7 +138,7 @@ export function usePrefDefinitions() {
             ] as const
         )
         .filter(([_name, { subCategories }]) => subCategories.length > 0),
-    []
+    [preferencesVisibilityContext]
```

Comment:
Careful here. This created a bug or a performance problem.
`preferencesVisibilityContext` is re-created on every render.
Thus, useMemo is re-run on every render.
If you had ESLint enabled, it would have reported this as a bug - please spend some time looking into fixing your IDE's ESLint issues as ESLint would be crucial for catching bugs once I'm not here to review every line of code